Mage.Sets/src/mage/cards/s/ShareTheSpoils.java

```diff
@@ -0,0 +1,311 @@
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ENTERS_THE_BATTLEFIELD ||
+                event.getType() == GameEvent.EventType.LOST || // Player conceedes
+                event.getType() == GameEvent.EventType.LOSES;  // Player loses by all other means
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        // The card that entered the battlefield was this one
+        if (event.getType() == GameEvent.EventType.ENTERS_THE_BATTLEFIELD) {
+            return event.getTargetId().equals(getSourceId());
+        }
+
+        // A player lost the game
+        return true;
+    }
+
+    @Override
+    public TriggeredAbility copy() {
+        return new ShareTheSpoilsExileETBAndPlayerLossAbility(this);
+    }
+
+    @Override
+    public String getRule() {
+        return "When {this} enters the battlefield or an opponent loses the game, " +
+                "exile the top card of each player's library.";
+    }
+}
+
+class ShareTheSpoilsExileCardFromEveryoneEffect extends OneShotEffect {
+
+    public ShareTheSpoilsExileCardFromEveryoneEffect() {
+        super(Outcome.Exile);
+    }
+
+    public ShareTheSpoilsExileCardFromEveryoneEffect(final ShareTheSpoilsExileCardFromEveryoneEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (source == null) { return false; }
+
+        PlayerList players = game.getState().getPlayersInRange(source.getControllerId(), game);
+        for (UUID playerId : players) {
+            Player player = game.getPlayer(playerId);
+            if (player == null) { continue; }
+
+            Card topLibraryCard = player.getLibrary().getFromTop(game);
+            if (topLibraryCard == null) { continue; }
+
+            player.moveCardsToExile(
+                    topLibraryCard,
+                    source,
+                    game,
+                    true,
+                    CardUtil.getExileZoneId(game, source),
+                    CardUtil.getSourceName(game, source)
+            );
+        }
+        return true;
+    }
+
+    @Override
+    public ShareTheSpoilsExileCardFromEveryoneEffect copy() { return new ShareTheSpoilsExileCardFromEveryoneEffect(this); }
+}
+
+//-- Play a card Exiled by Share the Spoils --//
+class ShareTheSpoilsPlayExiledCardEffect extends AsThoughEffectImpl {
+
+    ShareTheSpoilsPlayExiledCardEffect() {
+        super(AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, Duration.WhileOnBattlefield, Outcome.PutCardInPlay);
+        staticText = "During each player's turn, " +
+                "that player may play a land or cast a spell from among cards exiled with {this}, " +
+                "and they may spend mana as though it were mana of any color to cast that spell";
```

Comment:
you should add the text from the triggered ability to this effect so it's in the same line, then use `setRuleVisible` on the trigger so it doesn't show up separately